EIPS/eip-5450.md

```diff
@@ -13,67 +13,127 @@ requires: 3540, 3670, 4200, 4750
+   2. Compute new stack height after the instruction execution.
+2. Determine the list of successor instructions than can follow the current instructions:
+   1. The next instruction for all instructions other than terminating instructions and unconditional jump.
+   2. The target of a conditional or unconditional jump.
+3. For each successor instruction:
+   1. Check if the instruction is present in the code (i.e. execution must not "fall off" the code).
+   2. If the instruction does not have stack height recorded (visited for the first time):
+      1. Record the instruction stack height as the value computed in 1.2.
+      2. Add the instruction to the *worklist*.
+   3. Otherwise, check if the recorded stack height equals the value computed in 1.2.
+4. Determine the function maximum operand stack height:
+   1. Compute the maximum stack height as the maximum of all recorded stack heights.
+   2. Check if the maximum stack height does not exceed the limit (1024).
+
+The computational and space complexity of this pass is *O(len(code))*. Each instruction is visited at most once.
 
 ### Execution
 
-Given new deploy-time guarantees, EVM implementation is not required anymore to have run-time stack underflow check for each executed instruction. However implementations can keep it if they choose to do so.
+Given the deploy-time validation guarantees, an EVM implementation is not required anymore to have run-time stack underflow check for each executed instruction.
+
+Stack overflow check, on the other hand, is still required at run-time, because function execution can start at arbitrary (i.e. known only at run-time) stack height at `CALLF` instruction of a caller (i.e. each execution can be in arbitrary inner call frame). Validation algorithm examines only stack height changes relative to starting stack height of the function.
 
-Stack overflow check, on the other hand, is still required at run-time, because function execution can start at arbitrary (i.e. known only at run-time) stack height at `CALLF` instruction of a caller (i.e. each execution can be in arbitrary inner call frame). Verification algorithm examines only stack height changes relative to starting stack height of the function.
 
 ## Rationale
 
 ### Stack overflow check only in CALLF
 
-In the current proposal stack overflow checks are unchanged (i.e. are done for every instruction). However, we can provide more efficient variant where stack overflow check is performed only in `CALLF` instruction and uses called function's `max_stack_height` information for this. This decreases flexibility of an EVM program because `max_stack_height` corresponds to the worst-case control-flow path in the function.
-Moreover, the `max_stack_height` computed during validation must be stored alongside the code. This can be in the EOF itself or in implementation-defined format.
+In the current proposal stack overflow checks are unchanged (i.e. are done for every instruction).
```

Comment:
I think we decided to incorporate this into the EIP?

Reply:
Yes, this is now included.